src/components/views/DownloadModVersionSelectModal.vue

```diff
@@ -0,0 +1,82 @@
+            <br/>
+            <div class="columns is-vcentered">
+                <template v-if="currentVersion !== null">
+                    <div class="column is-narrow">
+                        <select class="select" disabled="true">
+                            <option selected>
+                                {{currentVersion}}
+                            </option>
+                        </select>
+                    </div>
+                    <div class="column is-narrow">
+                        <span class="margin-right margin-right--half-width"><span class="margin-right margin-right--half-width"/> <i class='fas fa-long-arrow-alt-right'></i></span>
+                    </div>
+                </template>
+                <div class="column is-narrow">
+                    <select class='select' :value="selectedVersion" @change="handleChange">
+                        <option v-for='(value, index) in versionNumbers' :key='index' :value='value'>
+                            {{value}}
+                        </option>
+                    </select>
+                </div>
+                <div class="column is-narrow">
+                    <span class="tag is-dark" v-if='selectedVersion === null'>
+                        You need to select a version
+                    </span>
+                    <span class="tag is-success" v-else-if='recommendedVersion === selectedVersion'>
+                        {{selectedVersion}} is the recommended version
+                    </span>
+                    <span class="tag is-success" v-else-if='versionNumbers[0] === selectedVersion'>
+                        {{selectedVersion}} is the latest version
+                    </span>
+                    <span class="tag is-danger" v-else-if='versionNumbers[0] !== selectedVersion'>
+                        {{selectedVersion}} is an outdated version
+                    </span>
+                </div>
+            </div>
+        </template>
+        <template v-slot:footer>
+            <button class="button is-info" @click="downloadThunderstoreMod">Download with dependencies</button>
+        </template>
+    </ModalCard>
+</template>
+
+<script lang="ts">
+
+import { mixins } from "vue-class-component";
+import { Component, Prop } from 'vue-property-decorator';
+
+import DownloadMixin from "../../components/mixins/DownloadMixin.vue";
+import ModalCard from '../ModalCard.vue';
+
+
+@Component({
+    components: {
+        ModalCard
+    },
+})
+export default class DownloadModVersionSelectModal extends mixins(DownloadMixin) {
+    @Prop({ required: true }) versionNumbers!: string[];
+    @Prop({ required: true }) selectedVersion!: string | null;
+    @Prop({ required: true }) currentVersion!: string;
+    @Prop({ required: true }) recommendedVersion!: string;
+    @Prop({ required: true }) downloadThunderstoreMod!: Function;
+    @Prop({ required: true }) setSelectedVersion!: Function;
```

Comment:
Moot point if my earlier comments check out, but these could use more accurate typing, e.g. `(selectedVersion: string) => void` for `setSelectedVersion`.